src/Mooc/Videos/Application/Remove/VideoRemover.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace CodelyTv\Mooc\Videos\Application\Remove;
+
+use CodelyTv\Mooc\Videos\Domain\VideoId;
+use CodelyTv\Mooc\Videos\Domain\VideoNotFound;
+use CodelyTv\Mooc\Videos\Domain\VideoRepository;
+
+final class VideoRemover
+{
+	private $repository;
+
+	public function __construct(VideoRepository $repository)
+	{
+		$this->repository = $repository;
+	}
+
+	public function __invoke(VideoId $id): void
+	{
+		$video = $this->repository->search($id);
+
+		if (null !== $video) {
+			$this->repository->save($video);
+		} else {
+			throw new VideoNotFound($id);
+		}
```

Comment:
Clausula de guarda:
if($video === null) {
	throw new VideoNotFound($id);
}
$this->repository->save($video);

Aunque ya de por sí esa función hace más de una cosa. Yo extraería la lógica de la búsqueda y el tratamiento de errores fuera.